### PR TITLE
Create/Edit Gradeable new gradeable model integration

### PR DESCRIPTION
**Pull request by KevinMackenzie**
(Part of #2317) The create/edit gradeable interface now exclusively uses the new gradeable model.  In this process i also removed the `AdminGradeable` model, which was only used as a vehicle for the parameters passed to the twig templates.  I also removed the `AdminGradeableView` class, which was doing a lot of controller work (fetching and constructing data).  This PR should not change any of the outward behavior of this page.

@saxocellphone Can you check to make sure the the autograding config changes you made are still completely functional.  I made some tweaks to it so it would play more nicely with the autosaving system.

### CODE REVIEW

**Comment by saxocellphone** [comment]
When I create a gradeable based on a current gradeable, and exit without making any changes (except for name and id), and try to grade a person, it gives this error

Notice: Undefined offset: 0 in /usr/local/submitty/site/app/views/grading/ElectronicGraderView.php on line 664
--
I took a look at that line, and it seems like it can't find the components. I tried it again on masters and it doesn't happen there.

**Comment by KevinMackenzie** [comment]
This is a semi-known issue where trying to grade a 'no submission' student on a gradeable with student-assigned pdf pages has issues.  The real issue is that I was setting the default pdf page to the wrong value

**Comment by saxocellphone** [comment]
The config uploads stuff works as intended

File: site/app/controllers/admin/AdminGradeableController.php [hunk 1]
```diff
@@ -285,7 +285,7 @@ private function newComponent(Gradeable $gradeable) {
             'text' => false,
             'peer' => false,
             'order' => -1,
-            'page' => -1
+            'page' => 0 // 0 means no page, -1 is student-assigned
```
Comment (CouleeApps) [inline site/app/controllers/admin/AdminGradeableController.php]: Use an enum? Or at least constants

**Pull request merged**